src/actions/csi_benchmark.py

```diff
@@ -0,0 +1,372 @@
+import contextlib
+import dataclasses
+import json
+import logging
+import os
+import shlex
+import shutil
+import subprocess
+import tempfile
+from pathlib import Path
+from typing import Optional
+
+import ops
+from charmhelpers.fetch.archiveurl import ArchiveUrlFetchHandler
+
+log = logging.getLogger(__name__)
+
+BENCH_HOME = Path("/home/ubuntu/kube-bench")
+BENCH_BIN = BENCH_HOME / "kube-bench"
+BENCH_CFG = BENCH_HOME / "cfg-ck"
+GO_PKG = "github.com/aquasecurity/kube-bench"
+RESULTS_DIR = "/home/ubuntu/kube-bench-results"
+
+# Remediation dicts associate a failing test with a Remedy to fix it.
+# Conservative fixes will probably leave the cluster in a good state.
+# Dangerous fixes will likely break the cluster.
+# Tuple examples:
+#  {'1.2.3': Remedy('manual -- we don't know how to auto fix this', None, None)}
+#  {'1.2.3': Remedy('cli', 'command to run', None)}
+#  {'1.2.3': Remedy('kv', 'snap', {cfg_key: value})}
+
+
+@dataclasses.dataclass
+class Remedy:
+    """Remedy class for benchmarking."""
+
+    type: str
+    command: Optional[str]
+    config: Optional[dict] = None
+
+    def run(self, test_num: int, test_remediation: str) -> int:
+        """Run the remedy command."""
+        if self.type == "manual":
+            log.info(
+                "Test %s: unable to auto-apply remedy.\n" "Manual steps:\n%s",
+                test_num,
+                test_remediation,
+            )
+        elif self.type == "cli":
+            cmd = shlex.split(self.command)
+            try:
+                out = subprocess.check_output(cmd)
+            except subprocess.CalledProcessError:
+                raise ActionError(f"Test {test_num}: failed to run: {cmd}")
+            else:
+                log.info("Test %s: applied remedy: %s\nOutput: %s", test_num, cmd, out)
+            return 1
+
+
+CONSERVATIVE = {
+    "0.0.0": Remedy("cli", 'echo "this is fine"', None),
+    # etcd (no known failures with a default install)
+    # k8s-control-plane (no known failures with a default install)
+    # k8s-worker (no known failures with a default install)
+}
+ADMISSION_PLUGINS = {
+    "enable-admission-plugins": (
+        "PersistentVolumeLabel",
+        "PodSecurityPolicy," "AlwaysPullImages",
+        "NodeRestriction",
+    )
+}
+DANGEROUS = {
+    "0.0.0": Remedy("cli", 'echo "this is fine"', None),
```

Comment:
Why do we have those dummy values in CONSERVATIVE and here?

Reply:
I don't know -- this comes from the original code
https://github.com/charmed-kubernetes/layer-cis-benchmark/blob/main/actions/cis-benchmark#L45

Reply:
these seemingly unnecessary values were originally introduced https://github.com/charmed-kubernetes/layer-cis-benchmark/pull/1